### ISSUE TITLE
"Invalid value" warnings on NaNs in box plot tests

### ISSUE THREAD

**Issue by gordonwoodhull** [dc-js/dc.js#628]
![image](https://cloud.githubusercontent.com/assets/1366709/3488066/1fbe45d2-04c5-11e4-8e37-e7100a9cda62.png)

**Comment by nordfjord** [comment]
This happens for brushes as well when `_x(_brushextent[0]) == NaN || _x(_brushextent[1]) == NaN`

pardon my psuedocode

**Comment by nordfjord** [comment]
and the funny thing is even if I surround a call to `dc.redrawAll()` in a `try/catch` statement the error bleeds through, so I'd like to implore you to consider rolling out a fix for this ASAP.

**Comment by gordonwoodhull** [comment]
It's a harmless warning in the console isn't it? 

Patches are always welcome!

**Comment by nordfjord** [comment]
It's a not so harmless warning in the console as it prevents any further redraws for some reason, at least for me, I'll see if I can get a demo up to showcase.

**Comment by gordonwoodhull** [comment]
Oh, I see; does the number of warnings become a performance problem? Or does it actually break at some point?

**Comment by gordonwoodhull** [comment]
The way that I have found to debug similar problems is to put a conditional breakpoint in the d3 code where it's about to emit the bad element. Then go up the stack and add a check to filter out or fix the bad data.

It doesn't look like the d3.box code this is based on has ever been updated, for better or worse.

https://github.com/d3/d3-plugins/tree/master/box

**Comment by nordfjord** [comment]
As soon as I hit this error my page refuses to function at all, however that might be due to the fact that I'm using a client side MVC framework for the application, and may in fact not be any fault of DC. I'll however try to wrap my head around some protection mechanisms and send a PR when I think I've found a solution.

**Comment by nordfjord** [comment]
Also I think we might be speaking of two different error sources here, I get errors from the brush drawing method, when the domain has two of the same values (in my case dates). And that only happens for me when I get have no data in my crossfilter, and therefore no data in my charts.

**Comment by gordonwoodhull** [comment]
Right, this is coming from the automated tests, and it produces dozens of hundreds of them without actually failing. So you're only seeing one error and then it stops?

An example would be great. Or you could see if an empty data set in the box plot tests causes failure:
https://github.com/dc-js/dc.js/blob/develop/spec/box-plot-spec.js

Once I have a failing test case it's usually pretty easy to come up with a fix. A browser warning is a bit harder to work with.

**Comment by gordonwoodhull** [comment]
The issue also appears when one of the boxes is compressed into a flat line:

http://stackoverflow.com/questions/34595043/how-to-set-y-domain-for-dc-js-box-plot-does-it-handle-negative-values
https://jsfiddle.net/gordonwoodhull/ufbwpwrh/
